### ISSUE TITLE
Adding list of undefined parameter names to the 'Bind parameters must not contain undefined' error message

### ISSUE THREAD

**Issue by theirongiant** [sidorares/node-mysql2#1706]
Currently the error message shown is:
`Error: Bind parameters must not contain undefined. To pass SQL NULL specify JS null`

I've added some code to list which named parameters are undefined so the error message becomes:
`Error: Bind parameters must not contain undefined (:id, :email). To pass SQL NULL specify JS null`

Contributing guidelines says it's best to start an issue on changes before creating a PR so thought I'd mention here to see if it's of interest.

**Comment by sidorares** [comment]
What do you want to change here?

Mysql does not have `null` type, and for that reason driver wants it to be explicit. If you want to send "unset" variables pass null instead, e.i `conn.execute(sql, [param1, param2, id ?? null, email ?? null]`)

**Comment by theirongiant** [comment]
I'm not changing how it works just adding more detail to the error message so that it'll list which of the named parameters in the query were undefined. Ideally in a live system you shouldn't really be seeing this error but if it does occur it's probably helpful to know which parameter was unexpectedly undefined.

**Comment by sidorares** [comment]
yes happy to see anything that would help users fix the problem
I assume you can only know name of the parameter when named placeholders are used? ( in case of array parameters we can still add something helpful, at least parameter index ). I'll see what metadata can be extracted from prepared statement info

**Comment by theirongiant** [comment]
Yeah the code I've written only works in the case of named parameters but happy to take a look to see if I can do something similar for array parameters as well.

**Comment by sidorares** [comment]
something similar to this message:

<img width="702" alt="image" src="https://user-images.githubusercontent.com/173025/203222180-8038678b-2054-4324-bd1c-57a7cbc87f26.png">

I remember some time ago that `reading 'test'` was not in the error message

**Comment by theirongiant** [comment]
Yeah that's the idea, got it working for array parameters now as well. I'll run some tests to make sure that there are no bugs and can create a PR for it.
